app/js/main.jsx

```diff
@@ -2,60 +2,58 @@ import React from 'react';
 import { render } from 'react-dom';
 import { createStore, applyMiddleware, compose } from 'redux';
 import { Router, Route, hashHistory } from 'react-router';
-import { syncHistoryWithStore } from 'react-router-redux';
 import { Provider } from 'react-redux';
 import thunk from 'redux-thunk';
+import { electronEnhancer } from 'redux-electron-store';
+import { ipcRenderer as ipc } from 'electron';
 
 import App from './components/pages/App';
 import Job from './containers/Job';
 import reducer from './reducers';
 import Auth from './containers/Auth';
 import DevTools from './util/devtools';
-import JobHistory from './components/JobHistory';
-import ArtifactDetail from './components/ArtifactDetail';
+import JobHistory from './containers/JobHistory';
+import ArtifactDetail from './containers/ArtifactDetail';
 
 import '!style-loader!css-loader!../css/main.css';
 
-let store;
 
-if (process.env.NODE_ENV === 'production') {
-    store = createStore(
-        reducer,
-        applyMiddleware(
-            thunk
-        )
-    );
-}
+const filter = {
+    plugins: true,
+    artifacts: true,
+    connection: true,
+    jobs: true,
+    currentDirectory: true,
+    tabState: true,
+    currentJob: true,
+    superTypes: true,
+    excludeUnfilteredState: true
+};
+
+const enhancer = process.env.NODE_ENV === 'production' ?
+    compose(applyMiddleware(thunk), electronEnhancer(filter)) :
+    compose(applyMiddleware(thunk), electronEnhancer(filter), DevTools.instrument());
+
+const store = createStore(reducer, enhancer);
 
 if (process.env.NODE_ENV === 'development') {
-    store = createStore(
-        reducer,
-        compose(
-            applyMiddleware(
-                thunk
-            ),
-            DevTools.instrument()
-        )
-    );
     if (module.hot) {
-        module.hot.accept('./reducers', () =>
-            store.replaceReducer(require('./reducers').default)
-        );
+        module.hot.accept('./reducers', () => {
+            ipc.sendSync('renderer-reload');
+            store.replaceReducer(require('./reducers'));
+        });
     }
 }
 
-const history = syncHistoryWithStore(hashHistory, store);
-
-
 render(
     <div>
         <Provider store={store}>
-            <Router history={history}>
+            <Router history={hashHistory}>
                 <Route path="/" component={App} />
                 <Route path="job/:pluginId/:actionType/:jobId" component={Job} />
                 <Route path="type=:type&uri=:uri&secret_key=:secret_key" component={Auth} />
```

Comment:
Now that we have the central store, any chance you could change the auth system to just dispatch straight to the store so we don't have this round-about fragment-catching.